python/python/bystro/proteomics/scripts/get_uniprot_id_gene_name_mapping.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+
+"""Download a list of uniprot_id / gene_name pairs from Uniprot and store the result on disk as csv."""
+
+import pandas as pd
+import tqdm
+from unipressed import UniprotkbClient  # type: ignore[import]
+
+from bystro.utils.config import BYSTRO_PROJECT_ROOT
+
+
+def get_uniprot_id_gene_name_mapping() -> pd.DataFrame:
+    """Get a dataframe associating each uniprot id with its cognate gene names."""
+    APPROXIMATE_TOTAL_RECORDS = 240_000
+    query_results = UniprotkbClient.search(
+        query="(organism_name:homo)",
+    )
+    uniprot_id_gene_name_mapping = []
+    for record in tqdm.trange(query_results.each_record(), total=APPROXIMATE_TOTAL_RECORDS):
+        genes = record.get("genes", [])  # type: ignore[attr-defined]
+        gene_names = []
+        for gene in genes:
+            if (gene_name := gene.get("geneName")) and (value := gene_name.get("value")):
+                gene_names.append(value)
+        if not gene_names:
+            gene_names = [None]
+        for gene_name in gene_names:
+            uniprot_id_gene_name_mapping.append(
+                (record["primaryAccession"], gene_name)  # type: ignore[index]
+            )
+    uniprot_id_gene_name_mapping_df = pd.DataFrame(
+        uniprot_id_gene_name_mapping, columns=["uniprot_accession", "gene_name"]
+    )
+    return uniprot_id_gene_name_mapping_df
+
+
+if __name__ == "__main__":
+    uniprot_id_gene_name_mapping_df = get_uniprot_id_gene_name_mapping()
+    output_filename = (
+        BYSTRO_PROJECT_ROOT / "python/python/bystro/proteomics" / "uniprot_id_gene_name_mapping.csv"
+    )
+    uniprot_id_gene_name_mapping_df.to_csv(output_filename, index=False)
```

Comment:
So the goal of this PR is to create this mapping file of uniprot IDs to gene names, so then it's on the user to replace them? Am I understanding that correctly? The reason I ask is because I think that users that will find it straightforward to replace uniprot IDs with Hugo gene names using a mapping file will likely find it straightforward to replace uniprot IDs with Hugo gene names on their own and users that may find that difficult could have trouble with the next step before their proteomics file is ready for proteomic/genomic combined analyses. But if the intention is just to have this available to make that conversion easier on our side if we need to then I'm happy with that.

Reply:
No, I don't think it should ever be incumbent upon the user to perform that mapping themselves.

Earlier we were worried that proteomics datasets might use Uniprot IDs for to reference peptides, whereas the annotation file uses HUGO symbols for gene names.  But it looks like Fragpipe is converting Uniprot identifiers to HUGO symbols for us already, so our concern was misplaced, which makes this PR superfluous.  @akotlar recommended PRing it anyway for now, so that if we later encounter a use case where such a mapping is necessary, the building blocks will already be in place.